fabric-game-rule-api-v1/src/main/java/net/fabricmc/fabric/mixin/gamerule/MinecraftServerMixin.java

```diff
@@ -16,18 +16,20 @@
 
 package net.fabricmc.fabric.mixin.gamerule;
 
-import java.util.function.BiConsumer;
-
+import com.llamalad7.mixinextras.injector.wrapmethod.WrapMethod;
+import com.llamalad7.mixinextras.injector.wrapoperation.Operation;
 import org.spongepowered.asm.mixin.Mixin;
-import org.spongepowered.asm.mixin.gen.Invoker;
 
 import net.minecraft.server.MinecraftServer;
-import net.minecraft.world.GameRules;
+import net.minecraft.world.rule.GameRule;
+
+import net.fabricmc.fabric.api.gamerule.v1.GameRuleEvents;
 
-@Mixin(GameRules.BooleanRule.class)
-public interface GameRulesBooleanRuleAccessor {
-	@Invoker
-	static GameRules.Type<GameRules.BooleanRule> invokeCreate(boolean initialValue, BiConsumer<MinecraftServer, GameRules.BooleanRule> changeCallback) {
-		throw new AssertionError("This shouldn't happen!");
+@Mixin(MinecraftServer.class)
+public abstract class MinecraftServerMixin {
+	@WrapMethod(method = "onGameRuleUpdated")
+	private <T> void invokeChangeCallbacks(GameRule<T> rule, T value, Operation<Void> original) {
+		original.call(rule, value);
+		GameRuleEvents.changeCallback(rule).invoker().onGameRuleUpdated(value, (MinecraftServer) (Object) this);
```

Comment:
I dont think this is the correct place to inject this, as you will miss all of the gamerule changes that happened without a server. (See TestWorldBuilderImpl)